dotnet/src/IntegrationTests/Connectors/OpenAI/OpenAIToolsTests.cs

```diff
@@ -460,6 +460,253 @@ public async Task ConnectorAgnosticFunctionCallingModelClassesCanBeUsedForAutoFu
         Assert.NotNull(getWeatherForCityFunctionCallResult.Result);
     }
 
+    [Fact]
+    public async Task ConnectorAgnosticFunctionCallingModelClassesCanBeUsedForManualFunctionCallingForStreamingAsync()
+    {
+        // Arrange
+        var kernel = this.InitializeKernel(importHelperPlugin: true);
+
+        var settings = new OpenAIPromptExecutionSettings() { ToolCallBehavior = ToolCallBehavior.EnableKernelFunctions };
+
+        var sut = kernel.GetRequiredService<IChatCompletionService>();
+
+        var chatHistory = new ChatHistory();
+        chatHistory.AddUserMessage("Given the current time of day and weather, what is the likely color of the sky in Boston?");
+
+        string? result = null;
+
+        // Act
+        while (true)
+        {
+            AuthorRole? authorRole = null;
+            var fccBuilder = new FunctionCallContentBuilder();
+            var textContent = new StringBuilder();
+
+            await foreach (var streamingContent in sut.GetStreamingChatMessageContentsAsync(chatHistory, settings, kernel))
+            {
+                textContent.Append(streamingContent.Content);
+                authorRole ??= streamingContent.Role;
+                fccBuilder.Add(streamingContent);
+            }
+
+            var functionCalls = fccBuilder.Build().ToArray();
```

Comment:
I would change the builder to be `IAsyncEnumerable` and used like this:

Preferably this could be an extension method to a Collection of `FunctionCallContent` 

i.e.: 

```csharp
public FunctionCallContentCollection : IEnumerable<FunctionCallContent> {}

public static class FunctionCallContentCollectionExtensions {
    public static IAsyncEnumerable<StreamingChatMessageContent> AddFromStreamingChatMessageContents(this FunctionCallContentCollection 
 collection, IAsyncEnumerable<StreamingChatMessageContent> extractFrom)
    //extraction logic here..
}


FunctionCallContentCollection functionCalls = [];
await foreach (var update in functionCalls.AddFromStreamingChatContents(sut.GetStreamingChatMessageContentsAsync(chatHistory, settings, kernel)) 
{ 
    textContent.Append(streamingContent.Content);
    authorRole ??= streamingContent.Role;
}

```

```suggestion
            var fccBuilder = new FunctionCallContentBuilder(sut.GetStreamingChatMessageContentsAsync(chatHistory, settings, kernel));
            var textContent = new StringBuilder();

            await foreach (var streamingContent in fccBuilder)
            {
                textContent.Append(streamingContent.Content);
                authorRole ??= streamingContent.Role;
            }

            var functionCalls = fccBuilder.FunctionCalls();
```

Reply:
It's a valid option, but I would like to understand its benefits before proceeding with it. In comparison to the option implemented in this PR that uses only one class (builder) for collecting function bits and pieces, the suggested option adds three classes: a builder that is IAsyncEnumerable, a function content collection, and the collection extension method(s) class to SK public API surface. I'm hesitant to go with it unless it provides additional benefits.

On a side note, I originally considered a similar design/option where there would be just an extension method, like this: `... IAsyncEnumerable<StreamingChatMessageContent> GetFunctionCalls(this IAsyncEnumerable<StreamingChatMessageContent> content, out FunctionCallContent[] functionCalls)` that could be elegantly used like this:
```csharp
FunctionCallContent[]? functionCalls = null;

await foreach (var streamingContent in sut.GetStreamingChatMessageContentsAsync(chatHistory, settings, kernel).GetFunctionCalls(out functionCalls))
{
    textContent.Append(streamingContent.Content);
    authorRole ??= streamingContent.Role;
}

if (functionCalls.Any())
{
    ...
}
```
However, C# does not allow having out parameters for iterators and async methods, resulting in CS1988 (Async methods cannot have ref, in, or out parameters) and CS1623 (Iterators cannot have ref, in, or out parameters).

Reply:
Discussed offline and decided to keep the builder.